include/lfp/lfp.h

```diff
@@ -202,6 +202,49 @@ int lfp_seek(lfp_protocol*, int64_t n);
 LFP_API
 int lfp_tell(lfp_protocol*, int64_t* n);
 
+/** Peels off the current protocol to expose the underlying one
+ *
+ * Retrieves the underlying protocol by releasing it from outer protocol's
+ * ownership. This effectively leaves the outer protocol in an undefined state.
+ * I.e. layers that supports peel calls `release()` on the protocol's
+ * unique_ptr.
+ *
+ * `lfp_peel()` is not implemented for leaf protocols such as the cfile
+ * protocol. Calling this function with a leaf protocol leaves the
+ * protocol intact and sets the reference pointer to nullptr.
+ *
+ * \param outer Outer protocol that will be peeled off
+ * \param inner Reference to the underlying protocol
+ *
+ * \retval LFP_OK Success
+ * \retval LFP_NOTIMPLEMENTED Layer does not support peel
+ */
+LFP_API
+int lfp_peel(lfp_protocol* outer, lfp_protocol** inner) noexcept(true);
+
+/** Peek into the underlying protocol
+ *
+ * Creates a pointer to the underlying protocol, while leaving the outer
+ * pointer in a valid state.  I.e. layers that supports peek calls `get()` on
+ * the protocol's unique_ptr.
```

Comment:
Do we really even need to tell the world that there's a unique_ptr there? Phrasing it like "this is similar to calling get() on a std::unique_ptr" should get the fundamental idea across, without actually mixing in implementation details.